plugin/docker/container.go

```diff
@@ -21,16 +21,21 @@ type container struct {
 	client *client.Client
 }
 
-func containerBase() *container {
+func containerBase(forInstance bool) *container {
```

Comment:
Not really a fan of this pattern. Am wondering if changing `forInstance` to `forChildSchemas` would make things clearer. Then again, maybe it's worth changing these to `containerTemplate` so that I can add the `NewEntryTemplate`/`NewEntry`. Note that we still need this to return a `container` object in order for Wash to get its type ID, its child schemas, etc.

I don't want to do a massive restructuring in this PR though, so any changes to what's here will be done once the initial metadata schema support's been added to `wash find`. Main reason is to get something working.

Reply:
I personally think having the argument name reflect what it does, rather than the intended use case (because those can change or their may be multiple use cases) makes more sense.

Reply:
I would lean towards separating the schema from instance creation be something the compiler can enforce. Having `MetadataSchema` be a method that `plugin.Entry` types have to implement, and then only calling it when interacting with `ChildSchemas`, might work.